components/brave_wallet/browser/filecoin_keyring.cc

```diff
@@ -113,27 +139,11 @@ std::string FilecoinKeyring::GetAddressInternal(HDKeyBase* hd_key_base) const {
   if (!hd_key_base)
     return std::string();
   HDKey* hd_key = static_cast<HDKey*>(hd_key_base);
-  auto uncompressed_public_key = hd_key->GetUncompressedPublicKey();
-  auto payload = BlakeHash(uncompressed_public_key, 20);
-  int protocol = static_cast<int>(mojom::FilecoinAddressProtocol::SECP256K1);
   // TODO(spylogsster): Get network from settings.
-  std::string network = "t";
-  return network + std::to_string(protocol) +
-         CreateAddressWithProtocol(payload, protocol);
-}
-
-std::string FilecoinKeyring::CreateAddressWithProtocol(
-    const std::vector<uint8_t>& payload,
-    int protocol_index) const {
-  std::vector<uint8_t> checksumPayload(payload);
-  checksumPayload.insert(checksumPayload.begin(), protocol_index);
-  auto checksum = BlakeHash(checksumPayload, 4);
-  std::vector<uint8_t> final(payload);
-  final.insert(final.end(), checksum.begin(), checksum.end());
-  std::string input(final.begin(), final.end());
-  std::string encoded_output = base::ToLowerASCII(
-      base32::Base32Encode(input, base32::Base32EncodePolicy::OMIT_PADDING));
-  return encoded_output;
+  return FilAddress::FromUncompressedPublicKey(
+             hd_key->GetUncompressedPublicKey(),
+             mojom::FilecoinAddressProtocol::SECP256K1, mojom::kFilecoinTestnet)
```

Comment:
When do we plan to update to use the real network? What do we miss still in order to use the real network instead of this hardcode testnet?

Reply:
I think we can do it already, will address it in next PRs